src/ControlSystem/Measurements/BNSCenterOfMass.hpp

```diff
@@ -91,6 +91,133 @@ void center_of_mass_integral_on_element(
+      tmpl::list<::Tags::Time, ::evolution::Tags::PreviousTriggerTime,
+                 ::Events::Tags::ObserverMesh<3>,
+                 ::Events::Tags::ObserverDetInvJacobian<Frame::ElementLogical,
+                                                        Frame::Inertial>,
+                 grmhd::ValenciaDivClean::Tags::TildeD,
+                 ::Events::Tags::ObserverCoordinates<3, Frame::Grid>>;
+
+  template <typename Metavariables, typename ArrayIndex,
+            typename ParallelComponent>
+  void operator()(const double time, const std::optional<double>& previous_time,
+                  const Mesh<3>& mesh,
+                  const Scalar<DataVector>& inv_det_jacobian,
+                  const Scalar<DataVector>& tilde_d,
+                  const tnsr::I<DataVector, 3, Frame::Grid>& x_grid,
+                  Parallel::GlobalCache<Metavariables>& cache,
+                  const ArrayIndex& array_index,
+                  const ParallelComponent* const /*meta*/,
+                  const ObservationValue& /*observation_value*/) const {
+    const LinkedMessageId<double> measurement_id{time, previous_time};
+
+    // Initialize integrals and perform local calculations
+    double mass_a = 0.;
+    double mass_b = 0.;
+    std::array<double, 3> first_moment_a = {0., 0., 0.};
+    std::array<double, 3> first_moment_b = {0., 0., 0.};
+    measurements::center_of_mass_integral_on_element(
+        &mass_a, &mass_b, &first_moment_a, &first_moment_b, mesh,
+        inv_det_jacobian, tilde_d, x_grid);
+
+    // Reduction
+    auto my_proxy =
+        Parallel::get_parallel_component<ParallelComponent>(cache)[array_index];
+    // We need a place to run RunCallback on... this does not need to be
+    // the control system using the CoM data.
+    auto& reduction_target_proxy = Parallel::get_parallel_component<
+        ControlComponent<Metavariables, tmpl::front<ControlSystems>>>(cache);
+    Parallel::ReductionData<
+        Parallel::ReductionDatum<LinkedMessageId<double>, funcl::AssertEqual<>>,
+        Parallel::ReductionDatum<double, funcl::Plus<>>,
+        Parallel::ReductionDatum<double, funcl::Plus<>>,
+        Parallel::ReductionDatum<std::array<double, 3>, funcl::Plus<>>,
+        Parallel::ReductionDatum<std::array<double, 3>, funcl::Plus<>>>
+        reduction_data{measurement_id, mass_a, mass_b, first_moment_a,
+                       first_moment_b};
+    Parallel::contribute_to_reduction<
+        measurements::PostReductionSendBNSStarCentersToControlSystem<
+            ControlSystems>>(std::move(reduction_data), my_proxy,
+                             reduction_target_proxy);
+  }
+
+  using is_ready_argument_tags = tmpl::list<::Tags::Time>;
+
+  template <typename Metavariables, typename ArrayIndex, typename Component>
+  bool is_ready(const double time, Parallel::GlobalCache<Metavariables>& cache,
+                const ArrayIndex& array_index,
+                const Component* const component) const {
```

Comment:
This function is redundant.  The same check is done before calling it, so it will always return true.

Reply:
Where is the check done? I looked at the Event base class, EventsAnd[Dense]Triggers, and RunEventsAnd[Dense]Triggers, and I didn't find it. Other events that the control system use have basically the same `is_ready` function.

Reply:
For dense triggers, `domain::CheckFunctionsOfTimeAreReadyPostprocessor` in `src/Domain/FunctionsOfTime/FunctionsOfTimeAreReady.hpp` will cause a check for any event with `needs_evolved_variables()` returning true.  For non-dense triggers, `domain::Actions::CheckFunctionsOfTimeAreReady` in the same file will have been run.

Some events had checks added before the general dense-trigger check existed and I didn't think to remove the now-redundant code when it was added.

Reply:
Ok I just changed this function to only return `true`. I think the larger change is beyond the scope of this PR